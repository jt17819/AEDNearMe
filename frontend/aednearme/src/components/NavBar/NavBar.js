import React from "react";
import {
  AppBar,
  Toolbar,
  CssBaseline,
  makeStyles,
  useTheme,
  useMediaQuery,
} from "@material-ui/core";
import { Link } from "react-router-dom";
import DrawerComponent from "./Drawer/Drawer.js";
import { MyMenu } from "../DropdownMenu/Menu.js";


export const NavBar = () => {
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("md"));

  const handleLogout = () => {
    sessionStorage.clear()
    alert('You have been logged out')
    window.location.reload()
  }

  return (
    <AppBar className={classes.appbar}
    >
      <CssBaseline />
      <Toolbar>
        {isMobile ? (
          <DrawerComponent />
        ) : (
          <div className={classes.root}>
            <div className="">
              <Link to="/" className={classes.logo}>AED <br /> nearMe </Link>
            </div>

            <div className={classes.navlinks}>

              <Link to="/" className={classes.link} >Home</Link>

              <MyMenu title="AED" className={classes.mymenu} menuItems={aedArray} />
              <MyMenu title="CPR" className={classes.mymenu} menuItems={cprArray} />
              <MyMenu title="Choking" className={classes.mymenu} menuItems={chokingArray} />
              <MyMenu title="Education Hub" className={classes.mymenu} menuItems={educationArray} />
              {
              sessionStorage.getItem('accessToken') ?
              <Link to="/" onClick={handleLogout} className={classes.link}>Logout</Link> :
              <Link to="/login" className={classes.link}>Login</Link>
            }

            </div>
   
          </div>
          
        )}
      </Toolbar>
    </AppBar>
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    marginLeft: theme.spacing(1),
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    width: "100%",
  },

  navlinks: {
    margin: theme.spacing(2),
    display: "flex",
    padding: "20px 0",
    borderBottom: "1px solid #fff"
  },

  logo: {
    flexGrow: "10",
    cursor: "pointer",
    textDecoration: "none",
    color: "#fff",
    fontSize: "40px",
    fontWeight: "bolder",
    lineHeight: "10px",
  },

  link: {
    textDecoration: "none",
    color: "#fff",
    fontWeight: "bold",
    
    // 64f227 blight green color
    // color: "#64f227",
    fontSize: "20px",
    marginLeft: theme.spacing(5),

    // hover navbar items changes color
    "&:hover": {
      color: "#000",
      borderBottom: "1px solid white",
    },
  },

// DOES NOT WORK WAKAWAKAWAKA Why???
  mymenu: {
    textDecoration: "none",
    color: "#fff",
    fontWeight: "bold",
    fontSize: "20px",
    marginLeft: theme.spacing(5),
    // hover navbar items changes color
    "&:hover": {
      color: "#000",
      borderBottom: "1px solid white",
    },
  },

  // NavBar
  appbar: {
    position: "static",
    // ba181b dark red colour
    background: '#ba181b',

    // black line above NavBar
    "&:before": {
      width: "100%",
      height: "10px",
      top: 0,
      backgroundColor: "#000",
      content: '" "',
      position: "absolute",
      transform: ` rotateZ(180deg)`,
    },

  },

}));


const aedArray = [
  {
    title: "How to use",
    path: '/AEDhowto'
  },
  {
    title: "Add NEW location",
    path: "/upload"
  }
]

const cprArray = [
  {
    title: "How to give",
    path: "/CPRhowto"
  }
]

const chokingArray = [
  {
    title: "How to help",
    path: '/choking'
  }
]

const educationArray = [
  {
    title: "QUIZ",
    path: "/quiz"
  },
  {
    title: "AED: How to use",
    path: "/aedhowto"
  },
  {
    title: "CPR: How to give",
    path: "/cprhowto"
  },
  {
    title: "Choking: How to help",
    path: "/choking"
  }
]

