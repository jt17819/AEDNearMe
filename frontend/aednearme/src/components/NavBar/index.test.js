import { NavBar } from "./NavBar";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import {BrowserRouter as Router} from "react-router-dom"


describe("NavBar", () => {
   let mockFunction;
   beforeEach(() => {
     mockFunction = jest.fn();
     // eslint-disable-next-line testing-library/no-render-in-setup
     render( <NavBar />, { wrapper: MemoryRouter } );
   });


   test("It renders the nav div", () => {
       let div = screen.getByRole("banner");
       expect(div).toBeInTheDocument();
     });

    //  test("it renders", () => {
    //    const navigation = screen.getByRole("nav");
    //    expect(navigation.textContent).toMatch("");
    //  });

    it('should take a snapshot', () => {
        // render(<NavBar />, { wrapper: MemoryRouter })
        const { asFragment } = render(<NavBar />, { wrapper: MemoryRouter })
        
        expect(asFragment(<NavBar />)).toMatchSnapshot()
      })

   });
