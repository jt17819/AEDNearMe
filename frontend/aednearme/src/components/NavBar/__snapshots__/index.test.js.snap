// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appbar-12 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="makeStyles-root-7"
      >
        <div
          class=""
        >
          <a
            class="makeStyles-logo-9"
            href="/"
          >
            AED 
            <br />
             nearMe 
          </a>
        </div>
        <div
          class="makeStyles-navlinks-8"
        >
          <a
            class="makeStyles-link-10"
            href="/"
          >
            Home
          </a>
          <div>
            <span
              aria-haspopup="true"
              data-testid="span"
              id="basic-button"
              style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
            >
              AED
            </span>
          </div>
          <div>
            <span
              aria-haspopup="true"
              data-testid="span"
              id="basic-button"
              style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
            >
              CPR
            </span>
          </div>
          <div>
            <span
              aria-haspopup="true"
              data-testid="span"
              id="basic-button"
              style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
            >
              Choking
            </span>
          </div>
          <div>
            <span
              aria-haspopup="true"
              data-testid="span"
              id="basic-button"
              style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
            >
              Education Hub
            </span>
          </div>
          <a
            class="makeStyles-link-10"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
